jax_galsim/exponential.py

```diff
@@ -0,0 +1,133 @@
+import jax.numpy as jnp
+
+from jax_galsim.gsobject import GSObject
+from jax_galsim.gsparams import GSParams
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+
+@_wraps(_galsim.Exponential)
+@register_pytree_node_class
+class Exponential(GSObject):
+
+    # The half-light-radius is not analytic, but can be calculated numerically
+    # by iterative solution of equation:
+    #     (re / r0) = ln[(re / r0) + 1] + ln(2)
+    _hlr_factor = 1.6783469900166605
+    _one_third = 1.0 / 3.0
+    _inv_twopi = 0.15915494309189535
+
+    _has_hard_edges = False
+    _is_axisymmetric = True
+    _is_analytic_x = True
+    _is_analytic_k = True
+
+    def __init__(
+        self, half_light_radius=None, scale_radius=None, flux=1.0, gsparams=None
+    ):
+
+        # Checking gsparams
+        gsparams = GSParams.check(gsparams)
+
+        if half_light_radius is not None:
+            if scale_radius is not None:
+                raise _galsim.GalSimIncompatibleValuesError(
+                    "Only one of scale_radius and half_light_radius may be specified",
+                    half_light_radius=half_light_radius,
+                    scale_radius=scale_radius,
+                )
+            else:
+                super().__init__(
+                    half_light_radius=half_light_radius, flux=flux, gsparams=gsparams
+                )
+
+        elif scale_radius is None:
+            raise _galsim.GalSimIncompatibleValuesError(
+                "Either scale_radius or half_light_radius must be specified",
+                half_light_radius=half_light_radius,
+                scale_radius=scale_radius,
+            )
+        else:
+            super().__init__(scale_radius=scale_radius, flux=flux, gsparams=gsparams)
+
+        self._r0 = self.scale_radius
+        self._inv_r0 = 1.0 / self._r0
+        self._norm = self.flux * Exponential._inv_twopi * self._inv_r0**2
+
+    @property
+    def scale_radius(self):
+        """The scale radius of the profile."""
+        if "half_light_radius" in self.params:
+            return self.params["half_light_radius"] / Exponential._hlr_factor
+        else:
+            return self.params["scale_radius"]
+
+    @property
+    def half_light_radius(self):
+        """The half-light radius of the profile."""
+        if "half_light_radius" in self.params:
+            return self.params["half_light_radius"]
+        else:
+            return self.params["half_light_radius"] * Exponential._hlr_factor
```

Comment:
:-| yes... this is embarassing.....

sigh... this shoud be caught by tests.... but the corresponding test cannot run yet with current implement tools.... which makes me wonder if there is a better way to do this....

The bug would be flagged eventually, when all exponential.py tests are able to run, but it means that in the meantime there can be code with bugs on the main branch... On the other hand, this is just because we are missing like the Position object which I have already implemented in a separate PR. 

Maybe we can say that in very early dev stage we will let this fly, once we have implemented the draw functions pretty much all the basic tests will be able to run, and we can do a review then of the tests that are missinng and find a way to manually run them then.

Reply:
ah! actually my jax jitting test would have caught that particular bug, if I had named it properly **facepalm**